src/logic/small.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import data.equiv.basic
+
+/-!
+# Small types
+
+A type is `w`-small if there exists an equivalence to some `S : Type w`.
+
+We provide a noncomputable model `shrink α : Type w`, and `equiv_shrink α : α ≃ shrink α`.
+
+A subsingleton type is `w`-small for any `w`.
+
+If `α ≃ β`, then `small.{w} α ↔ small.{w} β`.
+-/
+
+universes w v
+
+/--
+A type is `small.{w}` if there exists an equivalence to some `S : Type w`.
+-/
+class small (α : Type v) : Prop :=
+(equiv_small : ∃ (S : Type w) (e : α ≃ S), true)
```

Comment:
As earlier, is there a reason why `∃ (S : Type w), nonempty (α ≃ S)` is worse than this?

Reply:
Done.